src/Compilers/Core/Portable/PEWriter/MetadataWriter.PortablePdb.cs

```diff
@@ -831,5 +832,46 @@ private void EmbedSourceLink(Stream stream)
                 kind: _debugMetadataOpt.GetOrAddGuid(PortableCustomDebugInfoKinds.SourceLink),
                 value: _debugMetadataOpt.GetOrAddBlob(bytes));
         }
+
+        private void EmbedCompilerFlags(CommonPEModuleBuilder module)
+        {
+            byte[] bytes = new byte[0];
+
+            // TODO: Where to get compiler version? 
+
+            // TODO: Where to get source encoding
+
+            _debugMetadataOpt.AddCustomDebugInformation(
+                parent: EntityHandle.ModuleDefinition,
+                kind: _debugMetadataOpt.GetOrAddGuid(PortableCustomDebugInfoKinds.DeterministicCompilerFlags),
+                value: _debugMetadataOpt.GetOrAddBlob(bytes));
+        }
+
+        private void EmbedMetadataReferenceInformation(CommonPEModuleBuilder module)
```

Comment:
>EmbedMetadataReferenceInformation [](start = 21, length = 33)

Another thing that we'll need to handle for completeness is module references (`-addmodule` on command line). It's not really important scenario, but need to think about it at least. Can file an issue for now.

Reply:
Filed https://github.com/dotnet/roslyn/issues/44246